packages/client/components/GroupingKanbanColumn.tsx

```diff
@@ -115,6 +118,18 @@ const GroupingKanbanColumn = (props: Props) => {
     () => groups.filter((group) => group.reflections.length > 0),
     [groups]
   )
+
+  const allReflections = filteredReflectionGroups.flatMap(({reflections}) => reflections)
+  // only do magical sorting before users create groups
+  if (allReflections.length === filteredReflectionGroups.length) {
```

Comment:
this feels a little brittle. what if i group 1 & then ungroup it?
instead, let's put this in a hook that runs on initial render AND when nothing is grouped. that way the only time it'll run is if they refresh the initial state

Reply:
Let's say before sorting, the reflections are `A, B, A', B'`. Now you enter the group phase, we sort the cards for you so it looks like `A, A', B, B'`. You grabbed card `B`, group it with `B'` and ungroup it immediately. Don't you want the state resets to what it look like **before** you grabbing the card, i.e., `A, A', B, B'`? Wouldn't it feel bad if it goes back to `A, B, A', B'`?

Reply:
if the drop location of `B` is deterministic, you're absolutely right!
With the new subcolumns feature, when you ungroup `B` it could wind up in the first or 2nd column. if it got dropped & went to the 2nd column, it'd feel bad if it then magically teleported to the first column.

alternatively, let's say you group all 4 together, then ungroup all 4 in a random order. that ungrouping order might be `B', A', A, B`. It'd feel bad if they all got resorted once the last card got ungrouped.

Reply:
Alright makes sense! Technical question though: how do I achieve the effect of "running sth only **before** first render"? I cannot use `useEffect` as it will run **after** the render. `useState` feels weird too. It looks like something that should be inside `componentWillMount` but it seems there's no equivalent hooks for it.

Reply:
big picture, a functional component is the same as the `render` method of a class component, so everything inside the function gets called on render, never before. componentWillMount has been deprecated because in the future, react be doing some work, determine it won't finish that work within 14ms, and put it on pause, which could cause a whole lot of sadness if that hook got called but the component then had to wait for the next animation frame to run.

for this case, i'd suggest `useMemo` with no deps so it just runs once.